dockstore-webservice/src/main/java/io/dockstore/webservice/resources/EntryResource.java

```diff
@@ -381,6 +385,31 @@ public List<VersionVerifiedPlatform> getVerifiedPlatforms(@Parameter(hidden = tr
         return verifiedVersions;
     }
 
+    @GET
+    @Path("/{entryId}/syncing")
```

Comment:
Nit-picking, and we could always tweak the labeling in the UI, but isn't this endpoint indicating whether the entry is receiving GitHub notifications? It's not the same as syncing, because there might be a missing or malformed .dockstore.yml.

Thinking about it more, I wonder if it's worth having more than a boolean (and see my other comment for additional complications). But we might want to have statuses that are something like:

1. Not getting GitHub event notifications
2. Getting GitHub event notifications but not syncing
3. Getting GitHub event notifications and syncing

On the one hand, it might be overkill, on the other hand, I think people get mixed up because they have the notifications working, but the .dockstore.yml is incorrect, and they don't realize that's the error.

Also, maybe relevant, maybe not, what do we do in the events table when an update is filtered out in the .dockstore.yml? Is there an event for it, and if so, is it success/fail, or do we not even record it?

Reply:
> Nit-picking, and we could always tweak the labeling in the UI, but isn't this endpoint indicating whether the entry is receiving GitHub notifications? It's not the same as syncing, because there might be a missing or malformed .dockstore.yml.

People probably have different ideas about exactly what "syncing" means.   I picked it because I thought it was an ok name, and felt like a synonym for GitHub passing us updates when the entry changed.  My choice was more focused on the attempt, and maybe yours is on the end result?  But we can call it whatever the team wants.  Something like `isGitHubAppInstalled` strips away almost all ambiguity.

> Thinking about it more, I wonder if it's worth having more than a boolean (and see my other comment for additional complications). But we might want to have statuses that are something like:
>
> 1. Not getting GitHub event notifications
> 2. Getting GitHub event notifications but not syncing
> 3. Getting GitHub event notifications and syncing
>
> On the one hand, it might be overkill, on the other hand, I think people get mixed up because they have the notifications working, but the .dockstore.yml is incorrect, and they don't realize that's the error.

Sure, in addition to the notion of the App being installed/not, we can notify the user that the update itself was successful/not.  However, that's beyond the original scope of the ticket, and we should probably spend a lot more time figuring out the latter, because errors in the `.dockstore.yml` will happen much more often, and in a much broader variety of ways, than the App being uninstalled.

And yes, although the App and .dockstore.yml are parts of the same update process, it's not clear we should combine the statuses, at least in terms of what the webservice responds with.  Whether or not updates are being triggered, and the outcome of an update being processed, they're not necessarily the same thing.   It really needs some intensive pondering.

Reply:
> Nit-picking, and we could always tweak the labeling in the UI, but isn't this endpoint indicating whether the entry is receiving GitHub notifications? It's not the same as syncing, because there might be a missing or malformed .dockstore.yml.

IMO, if the .dockstore.yml is invalid, it still counts as synced because the GitHub app is installed and it synced the invalid .dockstore.yml :slightly_smiling_face: But I agree that "sync" is perhaps too ambiguous, although the UI portion may be more clear on the intent

Reply:
I'm with Kathy on what counts as syncing, but wonder if the return type should not be an enum and the endpoint should be called something like `versionControlState`  for expand-ability.